include/benchmark/benchmark_api.h

```diff
@@ -674,8 +721,12 @@ class Fixture: public internal::Benchmark {
       this->TearDown(st);
     }
 
+    // These will be deprecated ...
```

Comment:
Actually the fixture I wrote (which was removed in the meantime) to benchmark the counter api depended on the non-const state. And counter setup is the main driver behind this change. Do you still want it as a separate PR?